### DISCUSSION TITLE
Access to a metadata structure to build markdown help

### DISCUSSION THREAD

**Discussion by miiitch** [Tyrrrz/CliFx#139] [Q&A] [answered]
Hi,

Is there a structure that contains all the commands and parameters that I can access thought a command ? I try to add a command like "myapp gen-markdown-help" to create a markdown file from the cli commands for documentation purpose.

Regards,

Michel

**Comment by Tyrrrz** [comment, marked as answer]
Hi. The structure of the app/commands is detailed by these types: https://github.com/Tyrrrz/CliFx/tree/master/CliFx/Schema

But they are internal by design since it would involve exposing too many implementation details to make them public. The future version of CliFx will not be using these types anymore, as it'll be running on source generators (no release date yet).

**Reply by miiitch** [reply]
Thanks for you faster-than-light answer! I'll try another way to write my doc.
